src/pip/_internal/req/req_install.py

```diff
@@ -777,18 +777,49 @@ def install(
 
         global_options = global_options if global_options is not None else []
         if self.editable:
-            install_editable_legacy(
-                install_options,
-                global_options,
-                prefix=prefix,
-                home=home,
-                use_user_site=use_user_site,
-                name=self.name,
-                setup_py_path=self.setup_py_path,
-                isolated=self.isolated,
-                build_env=self.build_env,
-                unpacked_source_directory=self.unpacked_source_directory,
-            )
+            if self.metadata_directory and self.metadata_directory.endswith('.dist-info'):
+
+                # XXX the metadata_directory passed to
+                # 'prepare_metadata_for_build_wheel' is the parent
+                metadata_directory = os.path.dirname(self.metadata_directory)
+
+                self.pep517_backend.build_editable(
+                    metadata_directory,
+                    scheme.platlib,
+                    {}
+                )
+
+                # create empty RECORD. To make install_unpacked_parsed_wheel work.
+                with open(os.path.join(self.metadata_directory, 'RECORD'), 'w+'):
+                    pass
+
+                install_unpacked_parsed_wheel(
```

Comment:
I don't understand the need for this. Why can't the backend just provide a wheel, which we install using the normal `install_wheel` route? We already have a lot of concepts and we're now adding an "unpacked parsed wheel" that's used to communicate between backend and frontend. That seems like a recipe for confusion...

Reply:
This is only the easiest way to copy files into site-packages with pip. Remember the older PEP 376 concept which is a directory with a *.dist-info. That concept also appears in the PEP 517 metadata hook. An unhappy accident that the *.data/ directory would technically work?

As for pip that function should be broken up even without editable installs. It has a TODO. I've split out the only part that requires a ZipFile. https://github.com/pypa/pip/pull/8154/files/fc5341993c184447805d412bb7a0cb92c28425c5#diff-81eaeaa2196a8c5382958f2d9f22b593R313

Reply:
Hmm, I notice that this doesn't populate RECORD when it's installed. Thought that was happening before, but maybe not.